diff -up Clementine-1.4.1-11-gcecc1c1b5/3rdparty/libprojectm/CMakeLists.txt.omv~ Clementine-1.4.1-11-gcecc1c1b5/3rdparty/libprojectm/CMakeLists.txt
--- Clementine-1.4.1-11-gcecc1c1b5/3rdparty/libprojectm/CMakeLists.txt.omv~	2025-03-18 00:10:22.976913540 +0100
+++ Clementine-1.4.1-11-gcecc1c1b5/3rdparty/libprojectm/CMakeLists.txt	2025-03-18 00:10:35.464643425 +0100
@@ -4,14 +4,6 @@ SET(CPACK_PACKAGE_DESCRIPTION_SUMMARY "T
 
 include(CheckCXXCompilerFlag)
 
-cmake_policy(SET CMP0005 OLD)
-
-# This policy was only added in cmake 2.8.4
-if (CMAKE_VERSION VERSION_GREATER "2.8.3")
-  cmake_policy(SET CMP0017 OLD)
-endif (CMAKE_VERSION VERSION_GREATER "2.8.3")
-
-
 set(USE_DEVIL OFF)
 set(USE_FBO ON)
 set(USE_FTGL OFF)
diff -up Clementine-1.4.1-11-gcecc1c1b5/CMakeLists.txt.omv~ Clementine-1.4.1-11-gcecc1c1b5/CMakeLists.txt
--- Clementine-1.4.1-11-gcecc1c1b5/CMakeLists.txt.omv~	2025-03-18 00:12:10.465677298 +0100
+++ Clementine-1.4.1-11-gcecc1c1b5/CMakeLists.txt	2025-03-18 00:12:13.189691617 +0100
@@ -1,8 +1,6 @@
 cmake_minimum_required(VERSION 3.0.0)
 project(clementine)
 
-cmake_policy(SET CMP0053 OLD)
-
 include(CheckCXXCompilerFlag)
 include(FindPkgConfig)
 include(cmake/C++11Compat.cmake)
