diff -urN Clementine-1.2.2/CMakeLists.txt Clementine-1.2.2-patched/CMakeLists.txt
--- Clementine-1.2.2/CMakeLists.txt	2014-02-07 01:35:36.000000000 +1100
+++ Clementine-1.2.2-patched/CMakeLists.txt	2014-03-04 01:52:12.928492021 +1100
@@ -203,6 +203,8 @@
   DEPENDS "Taglib 1.8" "TAGLIB_VERSION VERSION_GREATER 1.7.999"
 )
 
+optional_component(VK ON "Vk.com support")
+
 optional_component(AUDIOCD ON "Devices: Audio CD support"
   DEPENDS "libcdio" CDIO_FOUND
 )
@@ -268,6 +270,11 @@
   find_package(Qt4 REQUIRED QtDbus)
 endif ()
 
+if (HAVE_VK)
+  find_library(VREEN_LIBRARIES vreen)
+  find_library(VREENOAUTH_LIBRARIES vreenoauth)
+endif(HAVE_VK)
+
 # We can include the Qt definitions now
 include(${QT_USE_FILE})
 
diff -urN Clementine-1.2.2/data/data.qrc Clementine-1.2.2-patched/data/data.qrc
--- Clementine-1.2.2/data/data.qrc	2014-02-07 01:35:36.000000000 +1100
+++ Clementine-1.2.2-patched/data/data.qrc	2014-03-04 01:52:12.928492021 +1100
@@ -401,5 +401,23 @@
         <file>volumeslider-handle_glow.png</file>
         <file>volumeslider-handle.png</file>
         <file>volumeslider-inset.png</file>
+        <file>vk/add.png</file>
+        <file>vk/bookmarks.png</file>
+        <file>vk/delete.png</file>
+        <file>vk/discography.png</file>
+        <file>vk/download.png</file>
+        <file>vk/edit.png</file>
+        <file>vk/find.png</file>
+        <file>vk/group.png</file>
+        <file>vk/my_music.png</file>
+        <file>vk/play_alt.png</file>
+        <file>vk/playlist.png</file>
+        <file>vk/recommends.png</file>
+        <file>vk/remove.png</file>
+        <file>vk/upload.png</file>
+        <file>vk/user.png</file>
+        <file>vk/deactivated.gif</file>
+        <file>providers/vk.png</file>
+        <file>vk/link.png</file>
     </qresource>
 </RCC>
diff -urN Clementine-1.2.2/src/CMakeLists.txt Clementine-1.2.2-patched/src/CMakeLists.txt
--- Clementine-1.2.2/src/CMakeLists.txt	2014-02-07 01:35:36.000000000 +1100
+++ Clementine-1.2.2-patched/src/CMakeLists.txt	2014-03-04 01:52:12.928492021 +1100
@@ -1131,6 +1131,27 @@
     internet/boxsettingspage.ui
 )
 
+# Vk.com support
+optional_source(HAVE_VK
+  SOURCES
+    internet/vkservice.cpp
+    internet/vksettingspage.cpp
+    internet/vkurlhandler.cpp
+    globalsearch/vksearchprovider.cpp
+    internet/vkmusiccache.cpp
+		internet/vksearchdialog.cpp
+  HEADERS
+    internet/vkservice.h
+    internet/vksettingspage.h
+    internet/vkurlhandler.h
+    globalsearch/vksearchprovider.h
+    internet/vkmusiccache.h
+		internet/vksearchdialog.h
+  UI
+    internet/vksettingspage.ui
+		internet/vksearchdialog.ui
+)
+
 # Hack to add Clementine to the Unity system tray whitelist
 optional_source(LINUX
   SOURCES core/ubuntuunityhack.cpp
@@ -1197,6 +1218,10 @@
   Qocoa
 )
 
+if(HAVE_VK)
+    target_link_libraries(clementine_lib ${VREEN_LIBRARIES} ${VREENOAUTH_LIBRARIES})
+endif(HAVE_VK)
+
 if(ENABLE_VISUALISATIONS)
   target_link_libraries(clementine_lib ${LIBPROJECTM_LIBRARIES})
 endif(ENABLE_VISUALISATIONS)
diff -urN Clementine-1.2.2/src/config.h.in Clementine-1.2.2-patched/src/config.h.in
--- Clementine-1.2.2/src/config.h.in	2014-02-07 01:35:36.000000000 +1100
+++ Clementine-1.2.2-patched/src/config.h.in	2014-03-04 01:52:12.929492037 +1100
@@ -42,6 +42,7 @@
 #cmakedefine HAVE_SPOTIFY_DOWNLOADER
 #cmakedefine HAVE_STATIC_SQLITE
 #cmakedefine HAVE_UBUNTU_ONE
+#cmakedefine HAVE_VK
 #cmakedefine HAVE_WIIMOTEDEV
 #cmakedefine IMOBILEDEVICE_USES_UDIDS
 #cmakedefine TAGLIB_HAS_OPUS
diff -urN Clementine-1.2.2/src/core/metatypes.cpp Clementine-1.2.2-patched/src/core/metatypes.cpp
--- Clementine-1.2.2/src/core/metatypes.cpp	2014-02-07 01:35:36.000000000 +1100
+++ Clementine-1.2.2-patched/src/core/metatypes.cpp	2014-03-04 01:52:12.929492037 +1100
@@ -16,6 +16,10 @@
 #include "podcasts/podcast.h"
 #include "ui/equalizer.h"
 
+#ifdef HAVE_VK
+#include "internet/vkservice.h"
+#endif
+
 #ifdef HAVE_DBUS
 #include <QDBusMetaType>
 #include "core/mpris2.h"
@@ -72,6 +76,11 @@
   qRegisterMetaType<Subdirectory>("Subdirectory");
   qRegisterMetaType<QList<QUrl> >("QList<QUrl>");
 
+#ifdef HAVE_VK
+  qRegisterMetaType<MusicOwner>("MusicOwner");
+  qRegisterMetaTypeStreamOperators<MusicOwner>("MusicOwner");
+#endif
+
 #ifdef HAVE_DBUS
   qDBusRegisterMetaType<QImage>();
   qDBusRegisterMetaType<TrackMetadata>();
diff -urN Clementine-1.2.2/src/internet/internetmodel.cpp Clementine-1.2.2-patched/src/internet/internetmodel.cpp
--- Clementine-1.2.2/src/internet/internetmodel.cpp	2014-02-07 01:35:36.000000000 +1100
+++ Clementine-1.2.2-patched/src/internet/internetmodel.cpp	2014-03-04 01:52:12.929492037 +1100
@@ -56,6 +56,9 @@
 #ifdef HAVE_BOX
   #include "boxservice.h"
 #endif
+#ifdef HAVE_VK
+  #include "vkservice.h"
+#endif
 
 using smart_playlists::Generator;
 using smart_playlists::GeneratorMimeData;
@@ -108,6 +111,9 @@
 #ifdef HAVE_BOX
   AddService(new BoxService(app, this));
 #endif
+#ifdef HAVE_VK
+  AddService(new VkService(app, this));
+#endif
 }
 
 void InternetModel::AddService(InternetService *service) {
diff -urN Clementine-1.2.2/src/internet/searchboxwidget.cpp Clementine-1.2.2-patched/src/internet/searchboxwidget.cpp
--- Clementine-1.2.2/src/internet/searchboxwidget.cpp	2014-02-07 01:35:36.000000000 +1100
+++ Clementine-1.2.2-patched/src/internet/searchboxwidget.cpp	2014-03-04 01:52:12.929492037 +1100
@@ -59,6 +59,11 @@
   QApplication::sendEvent(ui_->filter, event);
 }
 
+void SearchBoxWidget::SetText(const QString &text)
+{
+    ui_->filter->setText(text);
+}
+
 void SearchBoxWidget::keyReleaseEvent(QKeyEvent* e) {
   switch (e->key()) {
     case Qt::Key_Escape:
diff -urN Clementine-1.2.2/src/internet/searchboxwidget.h Clementine-1.2.2-patched/src/internet/searchboxwidget.h
--- Clementine-1.2.2/src/internet/searchboxwidget.h	2014-02-07 01:35:36.000000000 +1100
+++ Clementine-1.2.2-patched/src/internet/searchboxwidget.h	2014-03-04 01:52:12.929492037 +1100
@@ -41,6 +41,7 @@
 
 public slots:
   void FocusOnFilter(QKeyEvent* e);
+  void SetText(const QString& text);
 
 protected:
   void keyReleaseEvent(QKeyEvent* e);
diff -urN Clementine-1.2.2/src/ui/mainwindow.cpp Clementine-1.2.2-patched/src/ui/mainwindow.cpp
--- Clementine-1.2.2/src/ui/mainwindow.cpp	2014-02-07 01:35:36.000000000 +1100
+++ Clementine-1.2.2-patched/src/ui/mainwindow.cpp	2014-03-04 01:52:12.930492052 +1100
@@ -21,6 +21,10 @@
 #include <cmath>
 #include <memory>
 
+#ifdef HAVE_VK
+# include "internet/vkservice.h"
+#endif
+
 #include <QCloseEvent>
 #include <QDir>
 #include <QFileDialog>
@@ -318,6 +322,10 @@
   connect(ui_->action_love, SIGNAL(triggered()), SLOT(Love()));
   connect(ui_->action_toggle_scrobbling, SIGNAL(triggered()), InternetModel::Service<LastFMService>(), SLOT(ToggleScrobbling()));
 #endif
+
+#ifdef HAVE_VK
+  connect(ui_->action_love, SIGNAL(triggered()), InternetModel::Service<VkService>(), SLOT(AddToMyMusicCurrent()));
+#endif
   connect(ui_->action_clear_playlist, SIGNAL(triggered()), app_->playlist_manager(), SLOT(ClearCurrent()));
   connect(ui_->action_remove_duplicates, SIGNAL(triggered()), app_->playlist_manager(), SLOT(RemoveDuplicatesCurrent()));
   connect(ui_->action_remove_from_playlist, SIGNAL(triggered()), SLOT(PlaylistRemoveCurrent()));
diff -urN Clementine-1.2.2/src/ui/settingsdialog.cpp Clementine-1.2.2-patched/src/ui/settingsdialog.cpp
--- Clementine-1.2.2/src/ui/settingsdialog.cpp	2014-02-07 01:35:36.000000000 +1100
+++ Clementine-1.2.2-patched/src/ui/settingsdialog.cpp	2014-03-04 01:52:12.930492052 +1100
@@ -76,6 +76,10 @@
 #  include "internet/boxsettingspage.h"
 #endif
 
+#ifdef HAVE_VK
+#  include "internet/vksettingspage.h"
+#endif
+
 #include <QAbstractButton>
 #include <QDesktopWidget>
 #include <QPainter>
@@ -175,6 +179,11 @@
 #endif
 
   AddPage(Page_Spotify, new SpotifySettingsPage(this), providers);
+
+#ifdef HAVE_VK
+  AddPage(Page_Vk, new VkSettingsPage(this), providers);
+#endif
+
   AddPage(Page_Magnatune, new MagnatuneSettingsPage(this), providers);
   AddPage(Page_DigitallyImported, new DigitallyImportedSettingsPage(this), providers);
   AddPage(Page_BackgroundStreams, new BackgroundStreamsSettingsPage(this), providers);
diff -urN Clementine-1.2.2/src/ui/settingsdialog.h Clementine-1.2.2-patched/src/ui/settingsdialog.h
--- Clementine-1.2.2/src/ui/settingsdialog.h	2014-02-07 01:35:36.000000000 +1100
+++ Clementine-1.2.2-patched/src/ui/settingsdialog.h	2014-03-04 01:52:12.930492052 +1100
@@ -84,6 +84,7 @@
     Page_Dropbox,
     Page_Skydrive,
     Page_Box,
+    Page_Vk
   };
 
   enum Role {
