diff -up Clementine-1.4.0rc2/3rdparty/qsqlite/qsql_sqlite.cpp.omv~ Clementine-1.4.0rc2/3rdparty/qsqlite/qsql_sqlite.cpp
--- Clementine-1.4.0rc2/3rdparty/qsqlite/qsql_sqlite.cpp.omv~	2023-11-02 21:35:07.049879338 +0100
+++ Clementine-1.4.0rc2/3rdparty/qsqlite/qsql_sqlite.cpp	2023-11-02 21:35:13.180932475 +0100
@@ -572,7 +572,7 @@ bool QSQLiteDriver::open(const QString &
             sharedCache = true;
     }
 
-    sqlite3_enable_shared_cache(sharedCache);
+//    sqlite3_enable_shared_cache(sharedCache);
 
     if (sqlite3_open_v2(db.toUtf8().constData(), &d->access, openMode, NULL) == SQLITE_OK) {
         sqlite3_busy_timeout(d->access, timeOut);
