diff -up Clementine-1.4.0rc2/3rdparty/libprojectm/CMakeLists.txt.omv~ Clementine-1.4.0rc2/3rdparty/libprojectm/CMakeLists.txt
diff -up Clementine-1.4.0rc2/3rdparty/qxt/CMakeLists.txt.omv~ Clementine-1.4.0rc2/3rdparty/qxt/CMakeLists.txt
--- Clementine-1.4.0rc2/3rdparty/qxt/CMakeLists.txt.omv~	2023-03-09 00:15:04.517491020 +0100
+++ Clementine-1.4.0rc2/3rdparty/qxt/CMakeLists.txt	2023-03-09 00:15:07.215511683 +0100
@@ -1,5 +1,5 @@
 cmake_minimum_required(VERSION 3.0.0)
-set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD 20)
 
 set(QXT-SOURCES
   qxtglobal.cpp
diff -up Clementine-1.4.0rc2/3rdparty/taglib/CMakeLists.txt.omv~ Clementine-1.4.0rc2/3rdparty/taglib/CMakeLists.txt
--- Clementine-1.4.0rc2/3rdparty/taglib/CMakeLists.txt.omv~	2023-03-09 00:15:11.826547005 +0100
+++ Clementine-1.4.0rc2/3rdparty/taglib/CMakeLists.txt	2023-03-09 00:15:14.067564174 +0100
@@ -1,5 +1,5 @@
 cmake_minimum_required(VERSION 3.0.0)
-set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD 20)
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-delete-non-virtual-dtor")
 
 set(TAGLIB_SOVERSION_CURRENT  17)
diff -up Clementine-1.4.0rc2/3rdparty/utf8-cpp/CMakeLists.txt.omv~ Clementine-1.4.0rc2/3rdparty/utf8-cpp/CMakeLists.txt
--- Clementine-1.4.0rc2/3rdparty/utf8-cpp/CMakeLists.txt.omv~	2023-03-09 00:14:55.172419468 +0100
+++ Clementine-1.4.0rc2/3rdparty/utf8-cpp/CMakeLists.txt	2023-03-09 00:14:59.612453459 +0100
@@ -1,2 +1,2 @@
 cmake_minimum_required(VERSION 3.0.0)
-set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD 20)
diff -up Clementine-1.4.0rc2/CMakeLists.txt.omv~ Clementine-1.4.0rc2/CMakeLists.txt
--- Clementine-1.4.0rc2/CMakeLists.txt.omv~	2023-03-09 00:15:15.531575391 +0100
+++ Clementine-1.4.0rc2/CMakeLists.txt	2023-03-09 00:15:19.166603248 +0100
@@ -3,6 +3,7 @@ project(clementine)
 cmake_minimum_required(VERSION 3.0.0)
 cmake_policy(SET CMP0053 OLD)
 
+set(CMAKE_CXX_STANDARD 20)
 include(CheckCXXCompilerFlag)
 include(FindPkgConfig)
 include(cmake/C++11Compat.cmake)
diff -up Clementine-1.4.0rc2/ext/clementine-tagreader/CMakeLists.txt.omv~ Clementine-1.4.0rc2/ext/clementine-tagreader/CMakeLists.txt
--- Clementine-1.4.0rc2/ext/clementine-tagreader/CMakeLists.txt.omv~	2023-03-09 00:38:36.021592821 +0100
+++ Clementine-1.4.0rc2/ext/clementine-tagreader/CMakeLists.txt	2023-03-09 00:40:27.501501573 +0100
@@ -1,3 +1,4 @@
+set(CMAKE_CXX_STANDARD 20)
 include_directories(${PROTOBUF_INCLUDE_DIRS})
 include_directories(${CMAKE_CURRENT_SOURCE_DIR})
 include_directories(${CMAKE_CURRENT_BINARY_DIR})
@@ -30,6 +31,7 @@ target_link_libraries(clementine-tagread
   Qt5::Core
   Qt5::Network
   z
+  absl_log_internal_check_op absl_log_internal_message
 )
 
 if(${CMAKE_SYSTEM_NAME} MATCHES "FreeBSD")
diff -up Clementine-1.4.0rc2/ext/libclementine-tagreader/CMakeLists.txt.omv~ Clementine-1.4.0rc2/ext/libclementine-tagreader/CMakeLists.txt
--- Clementine-1.4.0rc2/ext/libclementine-tagreader/CMakeLists.txt.omv~	2023-03-09 00:38:43.307652297 +0100
+++ Clementine-1.4.0rc2/ext/libclementine-tagreader/CMakeLists.txt	2023-03-09 00:38:47.252684495 +0100
@@ -1,3 +1,4 @@
+set(CMAKE_CXX_STANDARD 20)
 include_directories(${PROTOBUF_INCLUDE_DIRS})
 include_directories(${CMAKE_CURRENT_SOURCE_DIR})
 include_directories(${CMAKE_CURRENT_BINARY_DIR})
diff -up Clementine-1.4.0rc2/src/CMakeLists.txt.omv~ Clementine-1.4.0rc2/src/CMakeLists.txt
--- Clementine-1.4.0rc2/src/CMakeLists.txt.omv~	2023-03-09 00:48:42.161511367 +0100
+++ Clementine-1.4.0rc2/src/CMakeLists.txt	2023-03-09 00:48:54.453610697 +0100
@@ -1302,6 +1302,8 @@ target_link_libraries(clementine_lib
   ${CMAKE_THREAD_LIBS_INIT}
   ${SQLITE_LIBRARIES}
 
+  absl_log_internal_check_op absl_log_internal_message
+
   Qocoa
   z
 )
