diff -up clementine-1.3.1/src/core/mergedproxymodel.cpp.gcc7 clementine-1.3.1/src/core/mergedproxymodel.cpp
--- clementine-1.3.1/src/core/mergedproxymodel.cpp.gcc7	2015-12-15 13:13:23.000000000 +0100
+++ clementine-1.3.1/src/core/mergedproxymodel.cpp	2017-09-27 04:44:33.762545326 +0200
@@ -25,10 +25,6 @@
 
 #include <limits>
 
-// boost::multi_index still relies on these being in the global namespace.
-using std::placeholders::_1;
-using std::placeholders::_2;
-
 #include <boost/multi_index_container.hpp>
 #include <boost/multi_index/member.hpp>
 #include <boost/multi_index/hashed_index.hpp>
diff -up clementine-1.3.1/src/devices/giolister.cpp.gcc7 clementine-1.3.1/src/devices/giolister.cpp
--- clementine-1.3.1/src/devices/giolister.cpp.gcc7	2015-12-15 13:13:23.000000000 +0100
+++ clementine-1.3.1/src/devices/giolister.cpp	2017-09-27 05:04:23.537579629 +0200
@@ -27,6 +27,8 @@
 #include "core/logging.h"
 #include "core/signalchecker.h"
 
+#include <functional>
+
 using std::placeholders::_1;
 using std::placeholders::_2;
 using std::placeholders::_3;
