diff -urN clementine-1.2.0/src/internet/searchboxwidget.cpp clementine-1.2.0-patched/src/internet/searchboxwidget.cpp
--- clementine-1.2.0/src/internet/searchboxwidget.cpp	2012-10-25 16:49:50.000000000 +1100
+++ clementine-1.2.0-patched/src/internet/searchboxwidget.cpp	2013-10-28 20:44:45.955783660 +1100
@@ -42,7 +42,7 @@
                    tr("Configure %1...").arg(service_->name()),
                    service_, SLOT(ShowConfig()));
 
-  ui_->filter->setPlaceholderText(QString("Search on %1").arg(service_->name()));
+  ui_->filter->setPlaceholderText(QString(tr("Search on %1")).arg(service_->name()));
   connect(ui_->filter, SIGNAL(textChanged(QString)), SIGNAL(TextChanged(QString)));
 
   did_you_mean_ = new DidYouMean(ui_->filter, this);
diff -urN clementine-1.2.0/src/translations/ru.po clementine-1.2.0-patched/src/translations/ru.po
--- clementine-1.2.0/src/translations/ru.po	2013-10-13 11:28:17.000000000 +1100
+++ clementine-1.2.0-patched/src/translations/ru.po	2013-10-28 20:46:14.401263286 +1100
@@ -5627,6 +5627,11 @@
 msgid "stop"
 msgstr "Остановить"
 
+#: internet/searchboxwidget.cpp:45
+#, qt-format
+msgid "Search on %1"
+msgstr "Искать на %1"
+
 #: widgets/osd.cpp:116
 #, qt-format
 msgid "track %1"
