From 8d921c15beb2c5d5489f3eadd0ee2021ea5ca311 Mon Sep 17 00:00:00 2001
From: Jonas Kvinge <jonas@jkvinge.net>
Date: Thu, 25 Jul 2019 19:02:43 +0200
Subject: [PATCH] Fix memory leaks in database

---
 src/core/database.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/core/database.cpp b/src/core/database.cpp
index 0eec39f89..caa227b49 100644
--- a/src/core/database.cpp
+++ b/src/core/database.cpp
@@ -262,7 +262,7 @@ QSqlDatabase Database::Connect() {
   }
 
   // Find Sqlite3 functions in the Qt plugin.
-  StaticInit();
+  if (!sFTSTokenizer) StaticInit();
 
   {
 
@@ -666,7 +666,7 @@ void Database::BackupFile(const QString& filename) {
   sqlite3* source_connection = nullptr;
   sqlite3* dest_connection = nullptr;
 
-  BOOST_SCOPE_EXIT((source_connection)(dest_connection)(task_id)(app_)) {
+  BOOST_SCOPE_EXIT((&source_connection)(&dest_connection)(task_id)(app_)) {
     // Harmless to call sqlite3_close() with a nullptr pointer.
     sqlite3_close(source_connection);
     sqlite3_close(dest_connection);
